File: Pertemuan 12/App Client API With Login/Anggota.php
<?php
require_once 'database.php';
class Anggota 
{
    private $db;
    private $table = 'anggota';
    public $kode_anggota = "";
    public $nama_anggota = "";
    public $jenis_kelamin = "";
    public $prodi = "";
    public $no_tlp = "";
    public $alamat = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_anggota(int $kode_anggota)
    {
        $query = "SELECT * FROM $this->table WHERE kode_anggota = $kode_anggota";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_anggota`,`nama_anggota`,`jenis_kelamin`,`prodi`,`no_tlp`,`alamat`) VALUES ('$this->kode_anggota','$this->nama_anggota','$this->jenis_kelamin','$this->prodi','$this->no_tlp','$this->alamat')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nama_anggota = '$this->nama_anggota', jenis_kelamin = '$this->jenis_kelamin', prodi = '$this->prodi', no_tlp = '$this->no_tlp', alamat = '$this->alamat' 
        WHERE id_anggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_anggota($kode_anggota): int
    {
        $query = "UPDATE $this->table SET kode_anggota = '$this->kode_anggota', nama_anggota = '$this->nama_anggota', jenis_kelamin = '$this->jenis_kelamin', prodi = '$this->prodi', no_tlp = '$this->no_tlp', alamat = '$this->alamat' 
        WHERE kode_anggota = $kode_anggota";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id_anggota = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_anggota($kode_anggota): int
    {
        $query = "DELETE FROM $this->table WHERE kode_anggota = $kode_anggota";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: Pertemuan 12/App Client API With Login/Anggota.py
import requests
import json
class Anggota:
    def __init__(self):
        self.__id=None
        self.__kode_anggota = None
        self.__nama_anggota = None
        self.__jenis_kelamin = None
        self.__prodi = None
        self.__no_tlp = None
        self.__alamat = None
        self.__url = "http://localhost/appperpus/anggota_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_anggota(self):
        return self.__kode_anggota
        
    @kode_anggota.setter
    def kode_anggota(self, value):
        self.__kode_anggota = value
    @property
    def nama_anggota(self):
        return self.__nama_anggota
        
    @nama_anggota.setter
    def nama_anggota(self, value):
        self.__nama_anggota = value
    @property
    def jenis_kelamin(self):
        return self.__jenis_kelamin
        
    @jenis_kelamin.setter
    def jenis_kelamin(self, value):
        self.__jenis_kelamin = value
    @property
    def prodi(self):
        return self.__prodi
        
    @prodi.setter
    def prodi(self, value):
        self.__prodi = value
    @property
    def no_tlp(self):
        return self.__no_tlp
        
    @no_tlp.setter
    def no_tlp(self, value):
        self.__no_tlp = value
    @property
    def alamat(self):
        return self.__alamat
        
    @alamat.setter
    def alamat(self, value):
        self.__alamat = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_anggota(self, kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id_anggota']
            self.__kode_anggota = item['kode_anggota']
            self.__nama_anggota = item['nama_anggota']
            self.__jenis_kelamin = item['jenis_kelamin']
            self.__prodi = item['prodi']
            self.__no_tlp = item['no_tlp']
            self.__alamat = item['alamat']
        return data
    def simpan(self):
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nama_anggota":self.__nama_anggota,
            "jenis_kelamin":self.__jenis_kelamin,
            "prodi":self.__prodi,
            "no_tlp":self.__no_tlp,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_anggota(self, kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        payload = {
            "kode_anggota":self.__kode_anggota,
            "nama_anggota":self.__nama_anggota,
            "jenis_kelamin":self.__jenis_kelamin,
            "prodi":self.__prodi,
            "no_tlp":self.__no_tlp,
            "alamat":self.__alamat
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_anggota(self,kode_anggota):
        url = self.__url+"?kode_anggota="+kode_anggota
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
